gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy

```diff
@@ -108,6 +108,71 @@ class PluginTest {
         }
     }
 
+    @Test
+    void pluginAddsRightRepositories_noRepositorySet() {
+        project.buildscript {
+            dependencies {
+                classpath "com.android.tools.build:gradle:${projectDependencies.get("GRADLE_BUILD_TOOLS")}"
+                classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.0'
+            }
+        }
+
+        def manifest = project.file("src/main/AndroidManifest.xml")
+        manifest.parentFile.mkdirs()
+        manifest.text = '<manifest xmlns:android="http://schemas.android.com/apk/res/android"  package="com.realm.test"></manifest>'
+
+        project.apply plugin: 'com.android.application'
+        project.apply plugin: 'realm-android'
```

Comment:
[nicpick] Relying on the apply order of plugins is considered a bad practice. It would be better to be reactive to the other plugins instead. This way a user can apply the plugin using the plugin bloc or in whatever order they wish while maintaining the same behaviour. This has repercussions on how the plugin is architectured which would require some change. It's something to consider changing to improve user happiness.